### PR TITLE
feat: Change CLI backend to Typer

### PR DESCRIPTION
**Pull request by abhijeetSaroha**
Change CLI backend to typer.

Solves #12

### CODE REVIEW

**Comment by abhijeetSaroha** [comment]
The error in `text_to_speech()` comes from file `tts.py` at the time of saving the output.

All other commands work fine, as I have checked all except `text_to_speech()`.
image

**Comment by xmnlab** [comment]
@abhijeetSaroha I just fixed some issues on upstream/main .. and rebased your branch

**Comment by xmnlab** [comment]
I am going to play with your new implementation now

**Comment by xmnlab** [comment]
@abhijeetSaroha , playing with that I found just few issus that would fix your code here.

this is the diff from changes:

```diff

diff --git a/src/artbox/cli.py b/src/artbox/cli.py
index 69dde42..247624d 100644
--- a/src/artbox/cli.py
+++ b/src/artbox/cli.py
@@ -92,13 +92,13 @@ def text_to_speech(
         typer.Option(
             "--lang", help="Choose the language for audio generation"
         ),
-    ] = "",
+    ] = "en",
 ) -> None:
     """Convert text to speech."""
     args_dict = {
         "title": title,
         "text-path": text_path,
-        "output_path": output_path,
+        "output-path": output_path,
         "engine": engine,
         "lang": lang,
     }

``` 
after that, check if that fixes the issues you had locally as well, please.

**Comment by abhijeetSaroha** [comment]
@xmnlab , corrected that mistake.

File: src/artbox/cli.py [hunk 1]
```diff
@@ -1,135 +1,254 @@
...
+            "--engine",
+            help="Choose the text-to-speech engine (Options: edge-tts, gtts)",
+        ),
+    ] = "edge-tts",
+    lang: Annotated[
+        str,
+        typer.Option(
+            "--lang", help="Choose the language for audio generation"
+        ),
+    ] = "",
```
Comment (xmnlab) [inline src/artbox/cli.py]:
```suggestion
    ] = "en",
```

it is also necessary to define its default as "en"

Comment (abhijeetSaroha) [reply src/artbox/cli.py]: Done.

**Comment by xmnlab** [comment]
LGTM! thanks for working on that @abhijeetSaroha ! appreciate that. good job here.
waiting the CI to be green and I will merge this PR. thanks!

**Pull request merged**